### ISSUE TITLE
Roadmap for this project?

### ISSUE THREAD

**Issue by JZubero** [capacitor-community/barcode-scanner#59]
First of all, many thanks to @tafelnl  for the good work on this module :clap: :bowing_man: :rocket: :beers: 

Currently this module is probably the only reliable option, since the [old Cordova module repo](https://github.com/phonegap/phonegap-plugin-barcodescanner) has already been archived and the pure web-based solutions (e.g. [zxing-js](https://github.com/zxing-js/ngx-scanner)) are not really suitable for Capacitor/Ionic (among other things because of the [iOS WebRTC v>=14.3 issue](https://github.com/zxing-js/ngx-scanner/issues/271)).

We are currently using this module in a project and therefore somewhat depend on it. So it would be great to know what developments are planned, preferably with a rough timeline. Especially features like [Frame in Camera view](https://github.com/capacitor-community/barcode-scanner/issues/10) or [torch integration](https://github.com/capacitor-community/barcode-scanner/issues/34) are mission critical for us and it's not clear if or when this will be addressed.

Again, we are grateful for all the work that has already been done :pray: Now we just want to find out if we can stick with the module :slightly_smiling_face:

**Comment by jdiaz49** [comment]
First of all, thank you very much for this excellent plugin, it has been very helpful for a feature of the app I'm working on.

Currently we need to use the flashlight and the cordova plugin I use conflicts with barcode scanner.

If this functionality of the flashlight is implemented in barcode scanner it would be excellent because it would not need another plugin.

It would be great to know when this flashlight functionality would be released to negotiate with my product owner hehehe.

Thank you very much for these wonderful plugins.

**Comment by Zwimber** [comment]
It [seems](https://github.com/capacitor-community/barcode-scanner/issues/34) to be priority 2. I assume that means that it is on the roadmap. Not sure when it'll be completed. If you look at the mentioned pull request you can try and install that version to get it working with the flashlight.

**Comment by tafelnl** [comment]
Quick update: I updated the priority labels on the different issues. Priority 1 issues are either important or easy to fix. These will be implemented in ~1 month. Priority 2 labels I cannot say for sure now. But I'll try to fix it in ~2 months. I am very busy right now, so any help is very welcome!

**Comment by arnotixe** [comment]
> If you look at the mentioned pull request you can try and install that version to get it working with the flashlight.

I'm very interested in this feature, as my users work in construction and scan stuff where lighting is poor / no lighting… Maybe not the best place to ask, but what is the best way to install the feat/torch branch via npm somehow?

I googled and found this, that kind of works:

npm install git://github.com/capacitor-community/barcode-scanner#feat/torch

Only smallish problem is it has no dist/ folder
![image](https://user-images.githubusercontent.com/500938/130312795-ae363035-bea3-423b-abfc-512ac7eb0a5d.png)

Found out how to checkout the branch, build it (separately) and copy in the dist/ folder… But maybe this can be done automatically somehow?

Turns on the torch with enableTorch() in android!

Doesn't turn off with disableTorch() though, but stopScan() serves :smile:

**Comment by dkliemsch** [comment]
I've just created a new PR based on the latest code from the main branch adding the ability to enable, disable and toggle the torch on both android and iOS. There is also a method to get the current torch state. 
@tafelnl Would be awesome if you could find some time to have a look and maybe create a new version as installing without a new npm release is really a pain (due to the missing dist folder)

@arnotixe Just so you know: The dist folder normally gets published on npm during a new release, which is why it works if you install an official version via the npm registry. If you try to install a specific branch / PR etc. directly from github that dist folder is missing.

**Comment by ec-lmayr** [comment]
i think it would suffice to add a prepare npm script to build the package.
like here:
https://github.com/ec-lmayr/barcode-scanner/commit/71399f0282ba7aa59966e1df794cd77cf6aae99f

the you can install from github.

**Comment by andreagobetti** [comment]
Hi, is there an ETA for the torch capability to be published via npm? I'm trying to add the torch branch via git to my project but it fails the dependency check...

Does this feature need more work or it's ready to be published?

Thanks!

**Comment by Pacjonek** [comment]
> First of all, many thanks to @tafelnl for the good work on this module 👏 🙇‍♂️ 🚀 🍻
> 
> Currently this module is probably the only reliable option, since the [old Cordova module repo](https://github.com/phonegap/phonegap-plugin-barcodescanner) has already been archived and the pure web-based solutions (e.g. [zxing-js](https://github.com/zxing-js/ngx-scanner)) are not really suitable for Capacitor/Ionic (among other things because of the [iOS WebRTC v>=14.3 issue](https://github.com/zxing-js/ngx-scanner/issues/271)).
> 
> We are currently using this module in a project and therefore somewhat depend on it. So it would be great to know what developments are planned, preferably with a rough timeline. Especially features like [Frame in Camera view](https://github.com/capacitor-community/barcode-scanner/issues/10) or [torch integration](https://github.com/capacitor-community/barcode-scanner/issues/34) are mission critical for us and it's not clear if or when this will be addressed.
> 
> Again, we are grateful for all the work that has already been done 🙏 Now we just want to find out if we can stick with the module 🙂
Could you elaborate why pure-web based solutions are not really suitable for Capacitor? It seems that WebRTC issue has been solved.

**Comment by thegnuu** [comment]
As of now, there is no roadmap... I just became a new maintainer of this plugin and I will try to clean up things and merge the open PRs in the coming weeks. 

Once I am up2date what have to be done, whats already done and not merged aso. I will try to add some sort of roadmap and plan the additionally wished features.

**Comment by jorrit** [comment]
Thanks for your effort!

**Comment by thegnuu** [comment]
I was able to merge most of the open PRs today...
Still need to test all the changes next week, once this is done I will release `2.1.0`.

For now I will close this issue, since there are multiple inputs/questions which makes it kinda hard for me to handle.

If i missed something here or you have missing features/questions after the `2.1.0` release please open a new issue!

Thank you guys!
